app/jobs/bulk_update_job.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+class BulkUpdateJob < ApplicationJob
+  def perform(ids:, args:)
+    change_set_persister.buffer_into_index do |buffered_change_set_persister|
+      ids.each do |id|
+        resource = query_service.find_by(id: id)
+        change_set = DynamicChangeSet.new(resource).prepopulate!
+        attributes = {}.tap do |attrs|
+          attrs[:state] = "complete" if args[:mark_complete] && !resource.state.include?("complete")
+        end
+        change_set.validate(attributes)
```

Comment:
Should we check the value of this so it doesn't update it if there's a validation error? Maybe even have it throw an exception so this batch dies?

Reply:
yes!

Reply:
@tpendragon I just added this please take a look. Thanks!